cmd/influxd/run/server.go

```diff
@@ -458,6 +458,9 @@ func (s *Server) Open() error {
 	s.TSDBStore.WithLogger(s.Logger)
 	if s.config.Data.QueryLogEnabled {
 		s.QueryExecutor.WithLogger(s.Logger)
+	} else if s.config.Coordinator.LogQueriesAfter > 0 {
+		// Log long-running queries even if not logging all queries
+		s.QueryExecutor.TaskManager.Logger = s.Logger
```

Comment:
What's the reason for setting the Logger reference directly instead of using the `.WithLogger()` method?

Reply:
`.WithLogger` sets the the logger for the whole query executor, and sets the `TaskManager.Logger` as well.  Here, we only want to set the `TaskManager.Logger` because `QueryLogEnabled` is `false`.

We could create a `WithLogger` method for the `TaskManager` if we want that abstraction, but it does not exist currently.